### PR TITLE
[CuTeDSL] Fix dynamic shape args not passed to JIT kernel

### LINKED ISSUE #2794: [BUG] [cuteDSL] mark_compact_shape_dynamic fails to pass shape args to JIT kernel (resolves to 0 at runtime)
### Issue Description
Hi, I've encountered a bug where dynamic tensor shapes marked with `mark_compact_shape_dynamic` are not correctly marshalled to the device kernel. When using static shapes, the kernel receives the correct values.

**Behavior:**
- **With `mark_compact_shape_dynamic`**: Inside the `@cute.kernel`, the dynamic dimensions resolve to **0** and strides contain garbage values, even when passing a valid concrete runtime tensor.
- **Without `mark_compact_shape_dynamic` (Static Shape)**: The kernel correctly receives the shape values (e.g., `(128, 128)`).

**Steps to Reproduce:**
1. Create a template GPU tensor.
2. (Bug Case) Mark dimensions as dynamic using `mark_compact_shape_dynamic`.
3. Compile a JIT kernel that launches a device kernel.
4. Pass a concrete runtime tensor.
5. Inspect `tensor.shape` inside the device kernel via `cute.printf`.

**Reproduction Script:**

```python
import torch
import cutlass.cute as cute
from cutlass.cute.runtime import from_dlpack

# Device Kernel
@cute.kernel
def device_kernel(tensor: cute.Tensor):
    tidx, _, _ = cute.arch.thread_idx()
    if tidx == 0:
        # BUG: These prints show (0, 0) when using dynamic shape
        # CORRECT: Shows (128, 128) when using static shape
        cute.printf("Device Kernel >> Tensor: {}\n", tensor)
        cute.printf("Device Kernel >> Shape: ({}, {})\n", tensor.shape[0], tensor.shape[1])

# Host Launcher
@cute.jit
def launcher_func(tensor: cute.Tensor):
    # Launch kernel with 1 thread to inspect values
    device_kernel(tensor).launch(grid=(1,1,1), block=(1,1,1))

def main():
    M, N = 128, 128
    
    # 1. Prepare Template (GPU Tensor -> gmem)
    fake_input = torch.zeros(M, N, device='cuda', dtype=torch.bfloat16)
    template = from_dlpack(fake_input, assumed_align=8)
    
    # [BUG REPRODUCTION]
    # Uncommenting these lines causes the kernel to see shape (0, 0)
    # Commenting them out (Static Shape) makes it work correctly: (128, 128)
    template = template.mark_layout_dynamic(leading_dim=1)
    template = template.mark_compact_shape_dynamic(
        mode=1,
        stride_order=(0, 1),
        divisibility=16
    )
    
    print(f"Compiling with template: {template}")
    
    compiled_func = cute.compile(launcher_func, template)
    
    print("\nRunning compiled function...")
    input_real = torch.randn(M, N, device='cuda', dtype=torch.bfloat16)
    input_runtime = from_dlpack(input_real, assumed_align=8)
    
    compiled_func(input_runtime)
    torch.cuda.synchronize()

if __name__ == "__main__":
    main()
```

**Actual Output (Dynamic Shape - Bug):**
```text
Device Kernel >> Shape: (0, 0)
```

**Expected Output (Static Shape - Reference):**
```text
Device Kernel >> Shape: (128, 128)
```

Thanks.

### PR DESCRIPTION
**Pull request by Flink-ddd**
## Purpose

Fixes #2794.

When mark_layout_dynamic + mark_compact_shape_dynamic are applied to a template tensor, the dynamic mask is stored in the template's DLTensorWrapper. Runtime tensors created via from_dlpack have no such marking, causing build_memref_desc to pack zero shapes into the memref descriptor.

Fix: record marking calls on _Tensor and replay them onto runtime tensors in `CudaDialectJitCompiledFunction.__call__` before generating execution args.

## Test Plan
Reproducible script using user issue

## Test Result

**Before fix:**

```
Running compiled function...
Device Kernel >> Tensor: raw_ptr(0x00007ac1ea408000: bf16, gmem, align<8>) o (0,0):(32,1)
Device Kernel >> Shape: (0, 0)
```

**After fix:**

```
Running compiled function...
Device Kernel >> Tensor: raw_ptr(0x000076839e408000: bf16, gmem, align<8>) o (128,128):(128,1)=
Device Kernel >> Shape: (128, 128)
```

### CODE REVIEW

**Comment by Flink-ddd** [comment]
Hi @hwu36 , Could you please take a look when you have time? Thanks!

**Comment by hwu36** [comment]
@Observer007 and @depaulmillz

**Comment by Flink-ddd** [comment]
Hi @Observer007  @depaulmillz , just a friendly follow-up on this PR when you have a moment. Thanks!

**Comment by Flink-ddd** [comment]
Hi @hwu36 @Observer007 @depaulmillz, the branch now has new conflicts with main due to recent updates. I've resolved them, could you find a moment to review when available? Thanks!
